pkg/registry/core/service/storage/storage.go

```diff
@@ -126,35 +126,35 @@ func (r *StatusREST) Update(ctx context.Context, name string, objInfo rest.Updat
 	return r.store.Update(ctx, name, objInfo, createValidation, updateValidation, false, options)
 }
 
-// defaults fields that were not previously set on read. becomes an
-// essential part of upgrading a service
-func (r *GenericREST) defaultServiceOnRead(obj runtime.Object) error {
-	if !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) {
-		return nil
-	}
-
+// defaultOnRead sets interlinked fields that were not previously set on read.
+// We can't do this in the normal defaulting path because that same logic
+// applies on Get, Create, and Update, but we need to distinguish between them.
```

Comment:
the Decorator is called on Create, Get and Update too,
https://github.com/kubernetes/kubernetes/blob/ecfcd5fe596b2683d5419b8ce2460272889f2d94/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

I've done a quick test and it seems that indeed is called for each operation
We are going to call `NormalizeClusterIPs` twice, no?

Reply:
Decorator is called at the very end of every operation.  For Create that should be a no-op and and for Update that isn't workable.  It should be a no-op in Update.  If we really want it ONLY on read, we would need a different hook, which seems unlikely to be warranted.

Reply:
ahhh, got it svcreg.NormalizeClusterIPs(nil, service), we pass nil as first argument
Thanks